sdk/bpf/c/inc/sol/cpi.h

```diff
@@ -11,6 +11,28 @@
 extern "C" {
 #endif
 
+/**
+ * Maximum CPI instruction data size. 10 KiB was chosen to ensure that CPI
+ * instructions are not more limited than transaction instructions if the size
+ * of transactions is doubled in the future.
+ */
+static const uint64_t MAX_CPI_INSTRUCTION_DATA_LEN = 10240;
+
+/**
+ * Maximum CPI instruction accounts. 255 was chosen to ensure that instruction
+ * accounts are always within the maximum instruction account limit for BPF
+ * program instructions.
+ */
+static const uint8_t MAX_CPI_INSTRUCTION_ACCOUNTS = 255;
+
+/**
+ * Maximum number of account info structs that can be used in a single CPI
+ * invocation. A limit on account info structs is effectively the same as
+ * limiting the number of unique accounts. 64 was chosen to match the max
+ * number of locked accounts per transaction (MAX_TX_ACCOUNT_LOCKS).
+ */
+static const uint8_t MAX_CPI_ACCOUNT_INFOS = 64;
+
```

Comment:
Please, do not change sdk/bpf/c/inc/sol/*.h files directly. They are generated from sdk/bpf/c/inc/sol/inc/*.inc files.

Reply:
My bad, can we add them to the gitignore then? Or at least add a comment that says it's generated?

Reply:
The generated files are also committed to the repository, we don't want to ignore them. It's not ideal to commit the generated files, but the assumption is that the header files are changed very rarely and it wouldn't be too much overhead to maintain them in repository. The generated files are currently regenerated when `cargo-install-all.sh` script runs. There are comments in the generated files, however, these header are placed above the syscall declarations that are generated by the tool that converts .inc files to corresponding .h files.  I understand the current situation is vulnerable and error prone but at the moment I have no better solution because the syscalls hash codes need to be generated.

Reply:
Ok, that's fine. Thanks for explaining!